Realm/Tests/KVOTests.mm

```diff
@@ -0,0 +1,1461 @@
+    KVORecorder r(self, obj, @"obj");
+    KVORecorder r2(self, obj, @"obj.invalidated");
+    [self.realm deleteObjects:[KVOLinkObject1 objectsInRealm:self.realm where:@"TRUEPREDICATE"]];
+
+    if (NSDictionary *note = AssertNotification(r)) {
+        XCTAssertTrue([note[NSKeyValueChangeOldKey] isKindOfClass:[RLMObjectBase class]]);
+        XCTAssertEqualObjects(note[NSKeyValueChangeNewKey], NSNull.null);
+    }
+    AssertChanged(r2, @NO, NSNull.null);
+}
+
+- (void)testDeleteObjectInArray {
+    KVOLinkObject2 *obj = [self createLinkObject];
+    KVOLinkObject1 *linked = obj.obj;
+    [obj.array addObject:linked];
+    KVORecorder r(self, obj, @"array");
+    [self.realm deleteObject:linked];
+    AssertIndexChange(NSKeyValueChangeRemoval, [NSIndexSet indexSetWithIndex:0]);
+}
+
+- (void)testDeleteObjectsInArrayViaTableClear {
+    KVOLinkObject2 *obj = [self createLinkObject];
+    KVOLinkObject2 *obj2 = [self createLinkObject];
+    [obj.array addObject:obj.obj];
+    [obj.array addObject:obj.obj];
+    [obj.array addObject:obj2.obj];
+
+    KVORecorder r(self, obj, @"array");
+    [self.realm deleteObjects:[KVOLinkObject1 allObjectsInRealm:self.realm]];
+    AssertIndexChange(NSKeyValueChangeRemoval, ([NSIndexSet indexSetWithIndexesInRange:{0, 3}]));
+}
+
+- (void)testDeleteObjectsInArrayViaQueryClear {
+    KVOLinkObject2 *obj = [self createLinkObject];
+    KVOLinkObject2 *obj2 = [self createLinkObject];
+    [obj.array addObject:obj.obj];
+    [obj.array addObject:obj.obj];
+    [obj.array addObject:obj2.obj];
+
+    KVORecorder r(self, obj, @"array");
+    [self.realm deleteObjects:[KVOLinkObject1 objectsInRealm:self.realm where:@"TRUEPREDICATE"]];
+    AssertIndexChange(NSKeyValueChangeRemoval, ([NSIndexSet indexSetWithIndexesInRange:{0, 3}]));
+}
+
+- (void)testObserveInvalidArrayProperty {
+    KVOObject *obj = [self createObject];
+    RLMArray *array = obj.arrayCol;
+    XCTAssertThrows([array addObserver:self forKeyPath:@"self" options:0 context:0]);
+    XCTAssertNoThrow([array addObserver:self forKeyPath:RLMInvalidatedKey options:0 context:0]);
+    XCTAssertNoThrow([array removeObserver:self forKeyPath:RLMInvalidatedKey context:0]);
+}
+@end
+
+// Mutate a different accessor backed by the same row as the accessor being observed
+@interface KVOMultipleAccessorsTests : KVOPersistedObjectTests
+@end
+@implementation KVOMultipleAccessorsTests
+- (id)observableForObject:(id)value {
+    if (RLMObject *obj = RLMDynamicCast<RLMObject>(value)) {
+        RLMObject *copy = [[obj.objectSchema.accessorClass alloc] initWithRealm:obj.realm schema:obj.objectSchema];
+        copy->_row = obj->_row;
+        return copy;
+    }
+    else if (RLMArray *array = RLMDynamicCast<RLMArray>(value)) {
+        return array;
+    }
+    else {
+        XCTFail(@"unsupported type");
+        return nil;
+    }
+}
+
+- (void)testIgnoredProperty {
+    // ignored properties do not notify other accessors for the same row
```

Comment:
No, it's removing a test defined in the parent class that doesn't apply to this subclass.